src/05-derivatives/02-common-data-types.md

```diff
@@ -1,4 +1,115 @@
+JSON file is also REQUIRED.
+Each derivative type defines their own set of fields, but all of them
+share the following (non-required) ones:
+
+| **Key name** | **Description**                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            |
+| ------------ | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Description  | RECOMMENDED. Free-form natural language description of the nature of the file.                                                                                                                                                                                                                                                                                                                                                                                                                                             |
+| Sources      | OPTIONAL. A list of paths relative to dataset root pointing to the file(s) that were directly used in the creation of this derivative. For example, if a derivative A is used in the creation of another derivative B, which is in turn used to generate C in a chain of A->B->C, C should only list B in `Sources`, and B should only list A in `Sources`. However, in case both X and Y are directly used in the creation of Z, then Z should list X and Y in `Sources`, regardless of whether X was used to generate Y. |
+| RawSources   | OPTIONAL. A list of paths relative to dataset root pointing to the BIDS-Raw file(s) that were used in the creation of this derivative. When the derivative filename does not define a `space` keyword, the first entry of `RawSources` MUST be defined and it will define the `scanner` coordinate system that applies.                                                                                                                                                                                                    |
+
+### Examples
+
+Preprocessed `bold` NIfTI file in the original coordinate space of the
+original run. Please mind that in this case `RawSources` key is REQUIRED
```

Comment:
I propose to clarify this as "Please mind that in this case, since the `space` keyword is not part of the file name, the  `RawSources` key is REQUIRED (and also..."